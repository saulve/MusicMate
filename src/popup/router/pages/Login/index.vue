<template>
  <div class="login--box flex flex--column">
    <h2 class="text-center">Welcome to MusicMate!</h2>
    <button
      class="flex align--items align--self button button--lrg bkg-green color--white"
      @click="login"
    >
      <icon 
        name="brands/spotify" 
        class="color--white"/>Login with Spotify
    </button>
  </div>
</template>

<script>
import { LOGIN_TO_SPOTIFY } from "../../../../store/action-types";

export default {
  data() {
    return {};
  },
  methods: {
    login() {
      this.$store.dispatch( LOGIN_TO_SPOTIFY );
    }
  }
};
</script>
