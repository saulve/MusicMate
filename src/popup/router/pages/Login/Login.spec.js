// import { render, fireEvent, cleanup } from "@testing-library/vue";
import { createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import TestHelper from "../../../../common/services/test-helper";
import Login from "./index.vue";

const localVue = createLocalVue();

localVue.use( Vuex );

describe( "Login Page", () => {
  let actions;
  let store;
  let stubs;

  beforeEach( () => {
    actions = {
      loginToSpotify: jest.fn()
    };
    store = new Vuex.Store({
      namespaced: true,
      state: {},
      actions
    });
    stubs = ["icon"];
  });

  it( "must render the component", () => {
    const wrapper = TestHelper.mountComponent( Login, localVue, store, stubs );
    expect( wrapper.element ).toMatchSnapshot();
    const loginButton = wrapper.find( "button" );
    expect( loginButton.text() ).toBe( "Login with Spotify" );
  });

  it( "must call login action when login button is clicked", async () => {
    const wrapper = TestHelper.mountComponent( Login, localVue, store, stubs );
    wrapper.find( "button" ).trigger( "click" );
    expect( actions.loginToSpotify ).toHaveBeenCalled();
  });
});
