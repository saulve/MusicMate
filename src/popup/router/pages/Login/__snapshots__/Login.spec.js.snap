// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page must render the component 1`] = `
<div
  class="login--box flex flex--column"
>
  <h2
    class="text-center"
  >
    Welcome to MusicMate!
  </h2>
   
  <button
    class="flex align--items align--self button button--lrg bkg-green color--white"
  >
    <icon-stub
      class="color--white"
      name="brands/spotify"
    />
    Login with Spotify
  
  </button>
</div>
`;
